neo/Network/P2P/TaskManager.cs

```diff
@@ -24,7 +24,15 @@ private class Timer { }
 
         private readonly NeoSystem system;
         private const int MaxConncurrentTasks = 3;
-        private readonly HashSet<UInt256> knownHashes = new HashSet<UInt256>();
+
+        private readonly Dictionary<int, UInt256> knownHashes = new Dictionary<int, UInt256>();
```

Comment:
why not use `FIFOSet` ?

Reply:
Could be, @shargon. In the beginning I was going to use Fifo but I found it complex because of some except `ExceptWith`.

Could you make the change to FIFOSet or recommend how to handle this?

Reply:
Done